vm/src/pyobject.rs

```diff
@@ -1402,6 +1411,26 @@ impl TryFromObject for std::time::Duration {
 
 result_like::option_like!(pub PyArithmaticValue, Implemented, NotImplemented);
 
+impl PyArithmaticValue<PyObjectRef> {
+    pub fn from_object(vm: &VirtualMachine, obj: PyObjectRef) -> Self {
+        if obj.is(&vm.ctx.not_implemented) {
+            Self::NotImplemented
+        } else {
+            Self::Implemented(obj)
+        }
+    }
+}
+
+impl<T: TryFromObject> TryFromObject for PyArithmaticValue<T> {
+    fn try_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<Self> {
+        let arith = match PyArithmaticValue::from_object(vm, obj) {
+            PyArithmaticValue::Implemented(x) => Self::Implemented(T::try_from_object(vm, x)?),
+            PyArithmaticValue::NotImplemented => Self::NotImplemented,
+        };
+        Ok(arith)
```

Comment:
PyArithmaticValue implements common option functions. So this is possible:
```suggestion
        PyArithmaticValue::from_object(vm, obj).map(|v| T::try_from_object(vm, x)).transpose()
```

Reply:
`transpose()` on PyArithmaticValue returns a `Result<Option<T>, E>` though, I guess I could do an into() on it but this seemed less verbose than all that

Reply:
By thinking of usage of transpose, I think that's bug of `ResultLike`. Let me fix it.

Reply:
result-like 0.3.0 will have fixed signature. I will do it later if you don't want to do it here.